examples/gno.land/r/sys/validators/validators.gno

```diff
@@ -45,33 +49,63 @@ func removeValidator(address std.Address) {
 	}
 
 	// Validator removed, note the change
-	changes = append(changes, change{
+	ch := change{
 		blockNum: std.GetHeight(),
 		validator: validators.Validator{
 			Address:     val.Address,
 			PubKey:      val.PubKey,
 			VotingPower: 0, // nullified the voting power indicates removal
 		},
-	})
+	}
+
+	saveChange(ch)
 
 	// Emit the validator set change
 	std.Emit(validators.ValidatorRemovedEvent)
 }
 
+// saveChange saves the valset change
+func saveChange(ch change) {
+	id := getBlockID(ch.blockNum)
+
+	setRaw, exists := changes.Get(id)
+	if !exists {
+		setRaw = make([]change, 0, 1)
+	}
+
+	// Save the change
+	set := setRaw.([]change)
+	set = append(set, ch)
+
+	changes.Set(id, set)
```

Comment:
If I'm not mistaken, you simply need to call this in the `!exists` function. Then, you can update the slice, but I'm not entirely sure about what I'm saying since it involves a slice rather than a pointed object.

Reply:
I've simplified it a bit, so it's easier to follow:

[c946718](https://github.com/gnolang/gno/pull/2229/commits/c946718a8f44c24f48eaa18c2e659a12901d9aaf)